src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,129 @@
     public void put(K key, V value) {
-
+        if (keyExists(key)) {
+            updateValue(key, value);
+            return;
+        }
+        if (size + 1 > currentCapacity * DEFAULT_LOAD_FACTOR) {
+            tab = resize();
+        }
+        Node<K, V> node = new Node<>(key, value);
+        placeNodeToTable(node);
     }
 
     @Override
     public V getValue(K key) {
+        for (Node<K, V> node : tab) {
+            while (node != null) {
+                if (Objects.equals(node.key,key)) {
+                    return node.value;
+                }
+                node = node.next;
+            }
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private Node<K,V>[] resize() {
+        int oldCapacity = currentCapacity;
+        currentCapacity = oldCapacity * SIZE_MULTIPLICATOR;
+        Node<K, V>[] oldTab = tab;
+        tab = new Node[currentCapacity];
+        size = 0;
+        for (Node<K, V> node : oldTab) {
+            Node<K, V> currentNode = node;
+            while (currentNode != null) {
+                put(currentNode.key, currentNode.value);
+                currentNode = currentNode.next;
+            }
+        }
+        return tab;
+    }
+
+    private void placeNodeToTable(Node<K, V> node) {
+        int index = Math.abs(getHash(node.key) % currentCapacity);
+        if (tab[index] == null) {
+            tab[index] = node;
+        } else {
+            Node<K, V> currentNode = tab[index];
+            while (currentNode.next != null) {
+                currentNode = currentNode.next;
+            }
+            currentNode.next = node;
+        }
+        size++;
+    }
+
+    private void updateValue(K key, V value) {
+        for (Node<K, V> node : tab) {
+            if (node != null) {
+                if (Objects.equals(node.key,key)) {
+                    node.value = value;
+                    return;
+                }
+                Node<K, V> currentNode = node;
+                while (currentNode != null) {
+                    if (Objects.equals(currentNode.key, key)) {
+                        currentNode.value = value;
+                        return;
+                    }
+                    currentNode = currentNode.next;
+                }
+            }
+        }
+    }
+
+    private boolean keyExists(K key) {
+        if (size == 0) {
+            return false;
+        }
+        for (Node<K, V> node : tab) {
+            if (node != null) {
+                if (Objects.equals(node.key,key)) {
+                    return true;
+                }
+                Node<K, V> currentNode = node;
+                while (currentNode != null) {
+                    if (Objects.equals(currentNode.key,key)) {
+                        return true;
+                    }
+                    currentNode = currentNode.next;
+                }
+            }
+        }
+        return false;
+    }
```

Comment:
the same here. And this logic is repeated three times. I would recommend you  introduce a method, that returns Node by key instead